music21/opensheetmusicdisplay/osmd.py

```diff
@@ -0,0 +1,189 @@
+from music21.instrument import Piano
+from urllib.request import pathname2url
+import time, random, json, os
+
+
+class ConverterOpenSheetMusicDisplay(SubConverter):
+    '''
+    Converter for displaying sheet music inline
+    in an IPython notebook.
+    Uses: https://opensheetmusicdisplay.org/ for rendering
+    '''
+    registerFormats = ('osmd',)
+    registerShowFormats = ('osmd',) 
+
+    def show(self, obj, fmt, 
+            fixPartName=True, offline=True, div_id=None,
+            **keywords):
+        score = obj
+        if fixPartName:
+            score = self.addDefaultPartName(score)
+
+        if div_id is None:
+            # create unique reference to output div in case we wish to update it
+            div_id = self.getUniqueDivId()
+            # div contents should be replaced by rendering
+            display(HTML('<div id="'+div_id+'">loading OpenSheetMusicDisplay</div>'))
+
+
+        xml = open(score.write('musicxml')).read()
+        script = self.musicXMLToScript(xml, div_id, offline=offline)
+
+        display(Javascript(script))
+        return div_id
+
+    @staticmethod
+    def getUniqueDivId():
+        return "OSMD-div-"+ \
+                str(random.randint(0,1000000))+ \
+                "-"+str(time.time()).replace('.','-') # '.' is the class selector
+
+    @staticmethod
+    def musicXMLToScript(xml, div_id, offline=True):
+        
+        # print('xml length:', len(xml))
+        script_dir = os.path.join(os.path.dirname(__file__), 'opensheetmusicdisplay.0.3.1.min.js')
+
+        # script that will replace div contents with OSMD display
+        script = """
+        console.log("loadOSMD()");
+        function loadOSMD() { 
+            return new Promise(function(resolve, reject){
+
+                if (window.opensheetmusicdisplay) {
+                    console.log("already loaded")
+                    return resolve(window.opensheetmusicdisplay)
+                }
+                console.log("loading osmd for the first time")
+                // OSMD script has a 'define' call which conflicts with requirejs
+                var _define = window.define // save the define object 
+                window.define = undefined // now the loaded script will ignore requirejs
+                var s = document.createElement( 'script' );
+                // alternative to local file is using a CDN
+                // s.setAttribute( 'src', "https://cdn.jsdelivr.net/npm/opensheetmusicdisplay@0.3.1/build/opensheetmusicdisplay.min.js" );
+                //s.setAttribute( 'src', "{{script_dir}}" );
+                function oncompleted(){
+                    window.define = _define
+                    console.log("loaded OSMD for the first time",opensheetmusicdisplay)
+                    resolve(opensheetmusicdisplay);
+                };
+                {{script_command}}
+                
+            }) 
+        }
+        loadOSMD().then((OSMD)=>{
```

Comment:
Arrow functions aren't supported in iOS 9 which is still a good chunk of the audience.  If they are to be used then other ES6 terms should be used, such as "let" and "const" instead of var.  I'm okay with dropping Safari 9 as a supported system if it's made explicit, but I'm a little worried that this code hasn't been tested in a variety of browsers.

Reply:
Arrow functions are now fine.